### PR TITLE
Develop v3

### PR DESCRIPTION
**Pull request by haunv3**
What is purpose of the changes?

 - Improve performance by clear redundant code & hook not use
 - Add try catch to handle error when called api

### CODE REVIEW

File: src/hooks/useLoadTokens.ts [hunk 1]
```diff
@@ -33,23 +33,27 @@ export type LoadTokenParams = {
...
+    // reset native balances
+    cosmosTokens
+      .filter((t) => t.chainId === tokenInfo.chainId && !t.contractAddress)
+      .forEach((t) => {
+        amountDetails[t.denom] = '0';
+      });
+
+    Object.assign(
+      amountDetails,
+      Object.fromEntries(amountAll.filter((coin) => tokenMap[coin.denom]).map((coin) => [coin.denom, coin.amount]))
```
Comment (tucq88) [inline src/hooks/useLoadTokens.ts:50]: Let's make it nicer: 

```
const tokenAmount = amountAll
    .filter((coin) => tokenMap[coin.denom])
    .map((coin) => [coin.denom, coin.amount])

Object.assign(
      amountDetails,
      Object.fromEntries(tokenAmounts)
 )
```

File: babel.config.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = function (api) {
-  api.cache(process.env.NODE_ENV === 'production');
+  api.cache(true);
```
Comment (tucq88) [inline babel.config.js:2]: so we aim to cache always, even with development?

File: src/layouts/FuturePromotion.tsx [hunk 3]
```diff
@@ -17,14 +17,14 @@ const FuturePromotion: React.FC = () => {
     window.open(LEARN_MORE_LINK, '_blank');
   };
 
-  useEffect(() => {
-    const TIMEOUT_TIME = 5000;
-    const TIMEOUT_ID = setTimeout(() => {
-      setIsOpen(true);
-    }, TIMEOUT_TIME);
+  // useEffect(() => {
```
Comment (tucq88) [inline src/layouts/FuturePromotion.tsx:20]: If we don't need it, let's delete it.

File: src/pages/Balance/TransferConvertToken/index.tsx [hunk 4]
```diff
@@ -160,7 +156,8 @@ const TransferConvertToken: FC<TransferConvertProps> = ({
 
   const to = findToTokenOnOraiBridge(token, filterNetwork);
   const fromTokenFee = useTokenFee(token.prefix + token.contractAddress);
-  const toTokenFee = useTokenFee(to?.chainId === 'oraibridge-subnet-2' ? to?.denom : to?.prefix + to?.contractAddress);
+  const remoteTokenDenomTo = to && to.chainId === 'oraibridge-subnet-2' ? to.denom : to.prefix + to.contractAddress;
```
Comment (tucq88) [inline src/pages/Balance/TransferConvertToken/index.tsx:159]: Can we make `oraibridge-subnet-2` a constant?

File: src/pages/Balance/index.tsx [hunk 5]
```diff
@@ -163,11 +168,25 @@ const Balance: React.FC<BalanceProps> = () => {
       if (!window.Metamask.isTron(from.chainId)) {
         await window.Metamask.switchNetwork(from.chainId);
       }
+
+      // remaining tokens, we override from & to of onClickTransfer on index.tsx of Balance based on the user's token destination choice
+      // to is Oraibridge tokens
+      // or other token that have same coingeckoId that show in at least 2 chain.
+      let newToToken = to;
+      if (filterNetwork) {
+        let newToToken = findToTokenOnOraiBridge(from, filterNetwork);
```
Comment (tucq88) [inline src/pages/Balance/index.tsx:177]: why need `let` here when we already initalised variable above?

**Comment by tucq88** [comment]
https://gyazo.com/01f7149d76d1cf3990067d5a480e5b10 
--
can't connect Metamask on my local

**Review by trungbach — APPROVED** [review]

**Comment by tucq88** [comment]
https://gyazo.com/5fd6c6713d88b22ce8cde0e09b619564 
nếu anh cố cancel thì nó lại toy @haunv3

**Comment by trungbach** [comment]
> https://gyazo.com/5fd6c6713d88b22ce8cde0e09b619564 nếu anh cố cancel thì nó lại toy @haunv3

crash này do mình đang throw error khi connect keplr failed ạ, em mới đổi lại sẽ show toast ra thay vì throw như kia ạ https://github.com/oraichain/oraiswap-frontend/pull/403/commits/c0d44f4e48bc27b4f941a54a6a0f5c51d77a098e

**Review by trungbach — DISMISSED** [review]

**Review by trungbach — APPROVED** [review]

**Review by tucq88 — APPROVED** [review]
LGTM

**Pull request merged**